core/src/main/java/com/linecorp/armeria/server/HttpHeaderUtil.java

```
@@ -59,8 +59,14 @@ final class HttpHeaderUtil {
     private static final CharMatcher QUOTED_STRING_TRIMMER = CharMatcher.is('"');
 
     @VisibleForTesting
-    static final Function<String, @Nullable String> FORWARDED_CONVERTER =
-            value -> TOKEN_SPLITTER.split(value).get("for");
+    static final Function<String, @Nullable String> FORWARDED_CONVERTER = value -> {
+        try {
+            return TOKEN_SPLITTER.split(value).get("for");
+        } catch (IllegalArgumentException e) {
+            logger.warn("Failed to parse Forwarded header chunk {}", value, e);
+            return null;
```

I now feel like it's probably better to fail the request to keep existing behavior

---

Do you mean you want to withdraw this PR?

---

Had a quick discussion about this with @yzfeng2020 and he will update this PR to just return a 400 response instead of silently ignoring a bad value.

---

If an invalid address is found in a Forwarded header, a debug message is logged. I was wondering if we need to change the behavior for consistency.
https://github.com/line/armeria/blob/ab6a934b184bd96133f4dae47ea832351998ba65/core/src/main/java/com/linecorp/armeria/server/HttpHeaderUtil.java#L157-L163

---

thanks @ikhoon, I went through some well known lib, and they are more of ignoring the malformed items and also having the option of erroring out. I am wondering if we can keep it simple and do the same: since we have been ignoring the errors with invalid address, it seems more consistent with ignoring the wrong format entries. Doing that would also gives the graceful degradation.

@trustin lmk what do you think again, thanks!

---

> I am wondering if we can keep it simple and do the same: since we have been ignoring the errors with invalid address

The `Forwarded` header is not essential in processing requests, so I prefer to log debug messages as we currently do. If there is a request to customize the behavior, we can consider adding an option at that point.

---

> The Forwarded header is not essential in processing requests

Note that services which serve requests behind multiple proxies (e.g. proxy1, proxy2, etc..) may allow/disallow requests based on the header. (I remember implementing such logic a couple years ago)

Usually, empty `Forwarded` headers signify that a request was made directly and not through proxies, and is generally allowed. It may not be possible to distinguish malformed `Forwarded` headers with local requests with this change.

I do think preferably we may want to always throw and let users decide how to distinguish malformed `Forwarded` headers from `ServerErrorHandler`